core/src/main/java/bisq/core/api/CoreMoneroConnectionsService.java

```diff
@@ -18,6 +18,9 @@
 import javafx.beans.property.SimpleObjectProperty;
 import javax.inject.Inject;
 import javax.inject.Singleton;
+
+import java.io.IOException;
```

Comment:
I was able to decouple the two services by replacing the daemon's external interface to pick the local monero daemon's instance if it had started, see `getPreferredDaemon`.

I removed an unnecessary call to `updateDaemonInfo` during initialization since it is done on a timer and could be incorrect since the local node has not started yet. Finally, I added the `startMoneroNode` of CoreMoneroNodeService at the end of CoreMoneroConnectionsService's initialization in case the user had last started a local node (indicating preference to continue using the local node). 

The local node setting and daemon is cleared when `stopMoneroNode` is called, and CoreMoneroConnectionService will provided the daemon connection as before. This allows the front end test to be simplified since we no longer have to track the additional local node "connection".

I think these changes to CoreMoneroConnectionService are unavoidable, since it provides the interface to access the daemon for the rest of the application via `getDaemon`.

Reply:
Thanks for the detailed explanation, I think I finally get it. 

One issue was that I had to add the hard coded password in the local monero node order to match the connection otherwise once you switch to a local monero node, it becomes persisted and you cannot restore the original hard coded connection with username and password. Efforts to restore the original hard coded value were difficult because the connection manager doesn't allow setting connection unless the node actually responded (the external monero process would be offline during the test).